clang/test/CIR/CodeGen/atomic-runtime.cpp

```diff
@@ -120,7 +120,7 @@ bool atomic_compare_exchange_n(int* ptr, int* expected,
 // CHECK:   cir.case(default, []) {
 // CHECK:     %[[expected:.*]] = cir.load %[[expected_addr]] : !cir.ptr<!s32i>, !s32i
 // CHECK:     %[[desired:.*]] = cir.load %[[desired_var]] : !cir.ptr<!s32i>, !s32i
-// CHECK:     %old, %cmp = cir.atomic.cmp_xchg(%[[ptr]] : !cir.ptr<!s32i>, %[[expected]] : !s32i, %[[desired]] : !s32i, success = relaxed, failure = relaxed) align(4) : (!s32i, !cir.bool)
+// CHECK:     %old, %cmp = cir.atomic.cmp_xchg(%[[ptr]] : !cir.ptr<!s32i>, %[[expected]] : !s32i, %[[desired]] : !s32i, success = relaxed, failure = relaxed) syncscope(system) align(4) : (!s32i, !cir.bool)
```

Comment:
I don't see any LLVM changes in the 3 files, is `syncscope(system)` the default? Since this PR isn't NFC, we should test at least one change to the final LLVM output (a case that uses `single_thread`).

Reply:
Since current codegen only generates system syncscope, I add a lowering test. If we want to generate singlethread syncscope from source C/C++ code, we need to add more emit functions.